src/command/exit_command.rs

```diff
@@ -81,4 +89,38 @@ mod tests {
         let cmd = ExitCommand::new(vec![String::from("42")]);
         assert_eq!(cmd.code, 42);
     }
+
+    #[test]
+    fn previous_cmd_exit_code() {
+        let mut prompt = Prompt::create(context::default());
+        prompt
+            .context
+            .borrow_mut()
+            .env
+            .insert("?".to_string(), "1".to_string());
+
+        let mut cmd = ExitCommand::new(vec![]);
+        if let Err(code) = cmd.execute(&mut prompt) {
+            assert_eq!(code, 1)
+        }
```

Comment:
You actually want to assert that it's an `Err` rather than asserting it's 1 _if_ it's an `Err`.
```rust
let res = cmd.execute(&mut prompt);
assert!(res.is_err());
assert_eq!(1, res.unwrap_err());
```
Or:
```rust
let res = cmd.execute(&mut prompt);
assert_eq!(1, res.unwrap_err());
```

Reply:
Fixed